### PR TITLE
Order cancelled

### PR DESCRIPTION
**Pull request by Eternal-Flame085**
- Finished user story 29
- Created model method to cancel order
- Added cancel order link to orders page
- added controller method to cancel order

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -13,4 +13,13 @@ def grandtotal
   def quantity
     items.count
   end
+
+  def cancel_order
+    item_orders.where(status: 'fulfilled').each do |order|
+      order.item.update(inventory: (order.quantity + order.item.inventory))
+    end
+
```
Comment (araspaas) [inline app/models/order.rb:21]: need to ask about this one in the checkin on monday

File: spec/features/orders/cancel_spec.rb [hunk 2]
```diff
@@ -67,7 +68,9 @@
     expect(page).to have_content('Order is now cancelled')
 
     visit '/profile/orders'
-    
+
+    expect(new_order.item_orders.first.item.inventory).to eq(5)
+
```
Comment (JBelta) [inline spec/features/orders/cancel_spec.rb:73]: Thanks for catching that!

File: spec/features/users/show_spec.rb [hunk 3]
```diff
@@ -38,7 +38,6 @@
 
       click_link "My Orders"
       expect(current_path).to eq('/profile/orders')
-      save_and_open_page
```
Comment (araspaas) [inline spec/features/users/show_spec.rb:41]: good find deleting this

**Review by araspaas — APPROVED** [review]
Lookin good

**Review by SageOfCode — APPROVED** [review]
Looks good to merge

File: app/models/order.rb (hunk 1 above)
Comment (JBelta) [reply app/models/order.rb:21]: Just thinking to myself if should this be an order model method or an item  model method?

**Review by JBelta — APPROVED** [review]

**Pull request merged**